libraries/core_libs/consensus/src/pbft/pbft_manager.cpp

```diff
@@ -1129,17 +1178,56 @@ uint64_t PbftManager::getThreshold(PbftVoteTypes vote_type) const {
   }
 }
 
-size_t PbftManager::placeVote_(taraxa::blk_hash_t const &blockhash, PbftVoteTypes vote_type, uint64_t round,
-                               size_t step) {
+uint64_t PbftManager::getPbftSortitionThreshold(PbftVoteTypes vote_type, uint64_t pbft_period) const {
+  const uint64_t total_dpos_votes_count = final_chain_->dpos_eligible_total_vote_count(pbft_period);
+
+  switch (vote_type) {
+    case propose_vote_type:
+      return std::min<uint64_t>(NUMBER_OF_PROPOSERS, total_dpos_votes_count);
+    case soft_vote_type:
+    case cert_vote_type:
+    case next_vote_type:
+    default:
+      return std::min<uint64_t>(COMMITTEE_SIZE, total_dpos_votes_count);
+  }
+}
+
+size_t PbftManager::placeVote_(taraxa::blk_hash_t const &blockhash, PbftVoteTypes vote_type, uint64_t period,
+                               uint64_t round, size_t step) {
   if (!weighted_votes_count_) {
     // No delegation
     return 0;
   }
 
-  auto vote = generateVote(blockhash, vote_type, round, step);
-  const auto weight =
-      vote->calculateWeight(getDposWeightedVotesCount(), getDposTotalVotesCount(), getThreshold(vote_type));
+  uint64_t voter_dpos_votes_count = 0;
+  uint64_t total_dpos_votes_count = 0;
+  uint64_t pbft_sortition_threshold = 0;
+
+  try {
+    // TODO[1896]: final_chain_->dpos... ret values should be cached as it is heavily used and most of the time it
+    // returns the same value!
+    voter_dpos_votes_count = final_chain_->dpos_eligible_vote_count(period, node_addr_);
+    total_dpos_votes_count = final_chain_->dpos_eligible_total_vote_count(period);
+    pbft_sortition_threshold = getPbftSortitionThreshold(vote_type, period);
+
```

Comment:
again this could be check if it is not already pre-cached

Reply:
will be solved by https://github.com/Taraxa-project/taraxa-node/issues/1896